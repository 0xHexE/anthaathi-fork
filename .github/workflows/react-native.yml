name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
