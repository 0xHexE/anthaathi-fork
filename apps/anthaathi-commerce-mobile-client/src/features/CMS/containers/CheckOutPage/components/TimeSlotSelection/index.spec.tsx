import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import TimeSlot from './index';

describe('TimeSlot', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <TimeSlot
            title="Timeslot"
            timeSlots={[
              {
                key: '1',
                name: '09am - 12pm',
              },
              {
                key: '2',
                name: '01pm - 05pm',
              },
              {
                key: '3',
                name: '05pm - 10pm',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('timeSlot')).toBeTruthy();
  });

  it('should have PromoCode title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <TimeSlot
            title="Timeslot"
            timeSlots={[
              {
                key: '1',
                name: '09am - 12pm',
              },
              {
                key: '2',
                name: '01pm - 05pm',
              },
              {
                key: '3',
                name: '05pm - 10pm',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Timeslot')).toBeTruthy();
  });
});
