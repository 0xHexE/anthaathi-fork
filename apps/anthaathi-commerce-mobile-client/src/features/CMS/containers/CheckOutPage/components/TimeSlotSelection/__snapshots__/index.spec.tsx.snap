// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSlot should render the item 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
  testID="timeSlot"
>
  <Text
    style={
      Object {
        "color": "#364A15",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    Timeslot
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginVertical": 5,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F1F9F4",
          "borderRadius": 2,
          "height": 38,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginRight": 10,
          "width": 108,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#364A15",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        09am - 12pm
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F1F9F4",
          "borderRadius": 2,
          "height": 38,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginRight": 10,
          "width": 108,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#364A15",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        01pm - 05pm
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F1F9F4",
          "borderRadius": 2,
          "height": 38,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginRight": 10,
          "width": 108,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#364A15",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        05pm - 10pm
      </Text>
    </View>
  </View>
</View>
`;
